extensions/amp-sidebar/1.0/sidebar.js

```diff
@@ -128,6 +129,16 @@ function SidebarWithRef(
       return;
     }
 
+    const document = sidebarElement.ownerDocument;
+    if (side != Side.LEFT && side != Side.RIGHT) {
+      sideRef.current = isRTL(document) ? Side.RIGHT : Side.LEFT;
+    } else {
+      sideRef.current = side;
+    }
+    sidebarElement.classList.add(
+      sideRef.current === Side.LEFT ? classes.left : classes.right
```

Comment:
FMI: why add these classes imperatively and not declaratively as they were done so before? Typically I'd worry about duplicate calls to add (without symmetric use of remove), but I think in this case it is OK because the className is rewritten on each render.

Reply:
Right, since the `sideRef.current` is not calculated until the `useLayoutEffect` (we need the `ref` from the rendered output) we do not know if we are going to end up with `left` or `right` yet for the side, so we have to add it here instead of `declaratively` as was done before.

Regarding multiple `adds` I found that it actually does not add multiple instances of the same class even if it is called twice.  And yes, since we have an updated sidebar elements each time the sidebar is opened, the classList is constantly reset.